worlds/ff6wc/WorldsCollide/data/enemy_packs.py

```diff
@@ -171,10 +174,38 @@ def randomize_event_bosses(self):
                         required_statue_formations.add(formation)
                     else:
                         required_boss_formations.add(formation)
+                elif condition.NAME == bosses_condition_name and condition.count > min_boss_formations:
+                    # if this condition is a set number of bosses,
+                    # make sure we always get the highest amount for the minimum req # of formations
+                    min_boss_formations = condition.count
                 elif condition.NAME == dragon_condition_name:
                     required_dragon_formations.add(condition.dragon_formation)
                 elif condition.NAME == dragons_condition_name and condition.count > min_dragon_formations:
                     min_dragon_formations = condition.count
+        # amount of bosses needed is the minimum from above minus the required bosses + required statues
+        boss_formations_needed = min_boss_formations - (len(required_boss_formations) + len(required_statue_formations))
+        # if the extra required number needed non-zero
+        if boss_formations_needed > 0:
+            # initialize all boss formations set
+            all_boss_formations = set(bosses.normal_formation_name)
+            # remaining bosses is all bosses - required bosses | required statues
+            required_formations = set(required_boss_formations | required_statue_formations)
+            remaining_boss_formations = list(all_boss_formations - required_formations)
```

Comment:
```suggestion
            remaining_boss_formations = sorted(all_boss_formations - required_formations)
```
I can see this is also a problem that already existed with Dragon Formations below.
We should probably fix it there too.

We want to be able to reproduce the same seed if we seed the random number generator.
But a `set` doesn't have a reliable deterministic order in Python.
So when we turn this set into a list, it could be in a different order each time, even if we use the same seed, which means we won't get the same results.

Any time we do something with a `set` that relies on the order, it should be sorted to make sure the order is the same every time.

Reply:
Should all instances of list be replaced with sorted? There are places elsewhere within enemy_packs.py that are not in this area of the code.

Reply:
Only if the parameter is a set and if the order makes a difference.

Reply:
For example, in this line:
```python
dragon_packs = list(bosses.dragon_pack_name)
```
`bosses.dragon_pack_name` is a `dict`
`dict` does have reliable deterministic order - It doesn't have the same problem.